File: app/src/main/java/com/sebastianlundquist/snapshot/MainActivity.java
package com.sebastianlundquist.snapshot;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener, View.OnKeyListener {

	boolean signUpModeActive = true;
	TextView loginText;
	EditText userInput;
	EditText passwordInput;

	public void signUp(View view) {
		if (!userInput.getText().toString().matches("") && !passwordInput.getText().toString().matches("")) {
			if (signUpModeActive) {

			}
			else {

			}
		}
		else {
			Toast.makeText(this, "Username and password are required.", Toast.LENGTH_SHORT).show();
		}
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		setTitle(R.string.login_title);
		loginText = findViewById(R.id.loginText);
		loginText.setOnClickListener(this);
		userInput = findViewById(R.id.userInput);
		passwordInput = findViewById(R.id.passwordInput);
		ImageView logoView = findViewById(R.id.logoView);
		ConstraintLayout backgroundLayout = findViewById(R.id.backgroundLayout);
		logoView.setOnClickListener(this);
		backgroundLayout.setOnClickListener(this);
		passwordInput.setOnKeyListener(this);
	}

	@Override
	public void onClick(View view) {
		if (view.getId() == R.id.loginText) {
			Button signUpButton = findViewById(R.id.signUpButton);
			if (signUpModeActive) {
				signUpModeActive = false;
				signUpButton.setText(R.string.login);
				loginText.setText(R.string.or_sign_up);
			}
			else {
				signUpModeActive = true;
				signUpButton.setText(R.string.sign_up);
				loginText.setText(R.string.or_login);
			}
		}
		else if (view.getId() == R.id.logoView || view.getId() == R.id.backgroundLayout) {
			InputMethodManager inputMethodManager = (InputMethodManager)getSystemService(INPUT_METHOD_SERVICE);
			inputMethodManager.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
		}
	}

	@Override
	public boolean onKey(View view, int i, KeyEvent keyEvent) {
		if (i == KeyEvent.KEYCODE_ENTER && keyEvent.getAction() == KeyEvent.ACTION_UP) {
			signUp(view);
		}
		return false;
	}
}

File: app/src/main/java/com/sebastianlundquist/snapshot/MainActivity.kt
package com.sebastianlundquist.snapshot

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.*
import androidx.constraintlayout.widget.ConstraintLayout
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity(), View.OnClickListener, View.OnKeyListener {

    internal var signUpModeActive = true
    lateinit var loginText: TextView
    lateinit var emailInput: EditText
    lateinit var passwordInput: EditText
    lateinit var logoView: ImageView
    lateinit var backgroundLayout: ConstraintLayout
    lateinit var signUpButton: Button
    val mAuth = FirebaseAuth.getInstance()

    fun signUpOrLogin(view: View) {
        if (!emailInput.text.toString().matches("".toRegex()) && !passwordInput.text.toString().matches("".toRegex())) {
            if (signUpModeActive) {
                mAuth.createUserWithEmailAndPassword(emailInput.text.toString(), passwordInput.text.toString())
                        .addOnCompleteListener(this) { task ->
                            if (task.isSuccessful) {
                                login()
                            } else {
                                Toast.makeText(baseContext, "Authentication failed.", Toast.LENGTH_SHORT).show()
                            }
                        }
            } else {
                mAuth.signInWithEmailAndPassword(emailInput.text.toString(), passwordInput.text.toString())
                        .addOnCompleteListener(this) { task ->
                            if (task.isSuccessful) {
                                login()
                            } else {
                                Toast.makeText(baseContext, "Authentication failed.", Toast.LENGTH_SHORT).show()
                            }
                        }
            }
        } else {
            Toast.makeText(this, "Username and password are required.", Toast.LENGTH_SHORT).show()
        }
    }

    fun login() {
        val intent = Intent(this, SnapsActivity::class.java)
        startActivity(intent)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setTitle(R.string.login_title)
        loginText = findViewById(R.id.loginText)
        loginText.setOnClickListener(this)
        emailInput = findViewById(R.id.userInput)
        passwordInput = findViewById(R.id.passwordInput)
        logoView = findViewById(R.id.logoView)
        backgroundLayout = findViewById(R.id.backgroundLayout)
        logoView.setOnClickListener(this)
        backgroundLayout.setOnClickListener(this)
        passwordInput.setOnKeyListener(this)

        if (mAuth.currentUser != null) {

        }
    }

    override fun onClick(view: View) {
        if (view.id == R.id.loginText) {
            signUpButton = findViewById(R.id.signUpButton)
            if (signUpModeActive) {
                signUpModeActive = false
                signUpButton.setText(R.string.login)
                loginText.setText(R.string.or_sign_up)
            } else {
                signUpModeActive = true
                signUpButton.setText(R.string.sign_up)
                loginText.setText(R.string.or_login)
            }
        } else if (view.id == R.id.logoView || view.id == R.id.backgroundLayout) {
            val inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            inputMethodManager.hideSoftInputFromWindow(currentFocus!!.windowToken, 0)
        }
    }

    override fun onKey(view: View, i: Int, keyEvent: KeyEvent): Boolean {
        if (i == KeyEvent.KEYCODE_ENTER && keyEvent.action == KeyEvent.ACTION_UP) {
            signUpOrLogin(view)
        }
        return false
    }
}
